bin/viper-per-sample-cluster.py

```diff
@@ -240,32 +327,39 @@ def bedtools_fasta_dict(fasta_text):
     return fasta_dict
 
 
-def quality_summary_selection(checkv_summary):
+def selection(checkv_summary, viraldf):
     """Function to select sequences with duplication/high k-mer warnings as assessed by CheckV."""
     # Read in CheckV's quality_summary file
     df = pd.read_csv(checkv_summary, sep="\t", dtype={"warnings": "str"})
     # Fill NA in warning column with empty string
     df["warnings"] = df["warnings"].fillna(
         ""
     )  # TO DO: find faster way to get rid of Nan
-
     include = set()
     exclude = set()
+
     # Define warnings
     warnings = [
         "high kmer_freq may indicate large duplication",
         "contig >1.5x longer than expected genome length",
     ]
-    # Change quality_summary df in dictionary
+    # Create a list of all the unique contigs names in the "source_seq" column of viraldf
     df_dict = df.to_dict("records")
-    for row in df_dict:
-        # Select any provirus contig with duplication/longer than expected warning and add it to the 'exclude' set
-        if row["provirus"] == "Yes" or any(x in row["warnings"] for x in warnings):
-            exclude.add(row["contig_id"])
-
-        # Select all contigs with warnings and add it to the 'include' set
-        if any(x in row["warnings"] for x in warnings):
-            include.add(row["contig_id"])
+    if not viraldf is not None:
+        proviruses_source_seq_names = set(list(viraldf["source_seq"]))
+        # Change quality_summary df in dictionary
+        for row in df_dict:
+            # Select any provirus contig with duplication/longer than expected warning and add it to the 'exclude' set
+            if row["contig_id"] in proviruses_source_seq_names or any(
+                x in row["warnings"] for x in warnings
+            ):
+                exclude.add(row["contig_id"])
```

Comment:
I think you forgot to add all contigs with warnings to the 'include' set
```suggestion
                exclude.add(row["contig_id"])
            if any(x in row["warnings"] for x in warnings):
                include.add(row["contig_id"])
```